internal/services/automation/automation_webhook.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2018-06-30-preview/automation"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceAutomationWebhook() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceAutomationWebhookCreateUpdate,
+		Read:   resourceAutomationWebhookRead,
+		Update: resourceAutomationWebhookCreateUpdate,
+		Delete: resourceAutomationWebhookDelete,
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.WebhookID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: resourceAutomationWebhookCommonSchema(),
+	}
+}
+
+func resourceAutomationWebhookCommonSchema() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+
+		"expiry_time": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: suppress.RFC3339Time,
+			ValidateFunc:     validation.IsRFC3339Time,
+		},
+
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"runbook_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.RunbookName(),
+		},
+		"run_on": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"parameters": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"uri": {
+			Type:       pluginsdk.TypeString,
+			ConfigMode: schema.SchemaConfigModeAttr,
+			Optional:   true,
+			ForceNew:   true,
+			Computed:   true,
+			Sensitive:  true,
+		},
```

Comment:
should we validate this?

Reply:
Yes, good catch. What's easy to add is:  `IsURLWithHTTPorHTTPS`. I don't think that there is any official pattern/domain, that we can validate. IIRC API was accepting some invalid (not pointing to Azure) urls on creation (but not on update...)

Reply:
Done